plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastoreConfig.java

```diff
@@ -65,4 +69,25 @@ public void testExplicitPropertyMappings()
 
         assertFullMapping(properties, expected);
     }
+
+    @Test
+    public void testStatsCacheTtl()
+    {
+        // enabled by default
+        assertThat(new CachingHiveMetastoreConfig().getStatsCacheTtl()).isEqualTo(DEFAULT_STATS_CACHE_TTL);
+
+        // takes higher of the DEFAULT_STATS_CACHE_TTL or metastoreTtl if not set explicitly
+        assertThat(new CachingHiveMetastoreConfig()
+                .setMetastoreCacheTtl(new Duration(1, SECONDS))
+                .getStatsCacheTtl()).isEqualTo(DEFAULT_STATS_CACHE_TTL);
+        assertThat(new CachingHiveMetastoreConfig()
+                .setMetastoreCacheTtl(new Duration(1111, DAYS))
+                .getStatsCacheTtl()).isEqualTo(new Duration(1111, DAYS));
+
+        // explicit configuration is honored
+        assertThat(new CachingHiveMetastoreConfig()
+                .setStatsCacheTtl(new Duration(135, MILLISECONDS))
+                .setMetastoreCacheTtl(new Duration(1111, DAYS))
```

Comment:
Can we disallow this configuration ? I can't think of a good reason why someone would want statistics metadata ttl to be shorter than other metadata ttl.

Reply:
Yes we can. Neither I can think of a good reason why someone would want that. OTOH, these are separate configuration toggles, so unclear what we gain by disallowing certain configurations which just work otherwise.

Reply:
Disallowing it prevents users from making a mistake in the configuration and saves debugging time for us.

Reply:
There are two toggles 

- A `hive.metastore-cache-ttl`
- B `hive.metastore-stats-cache-ttl`

if B < A, we can call it a mistake and reject. Do we have guarantee this is a mistake?
Or we risk that we gonna need to spend time explaining why we thought this is a good idea to reject such configurations?

BTW I think this discussion isn't very important. The whole problem comes from the fact that separate ``hive.metastore-stats-cache-ttl`` toggle was introduced so that it can be enabled by default. This PR doesn't change anything with respect to this main issue -- being enabled by default; default behavior doesn't change. So I guess we're spending time about some edge case.

Reply:
discussed offline and concluded this will be as it is for now at least